### PR TITLE
feat: secondsToDissolveDelayDuration

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Replace static sns min. dissolve delay with dynamic prop.

## Note:
Slider wasn't updated because its destiny is not clear yet.

# Changes

- util `secondsToDissolveDelayDuration`

# Tests

- `secondsToDissolveDelayDuration`

# Screenshots
127 0 0 1_5173_neuron__u=s24we-diaaa-aaaaa-aaaka-cai neuron=9d49faba9fd2d31a5705db307bcef7ae0ef1ba2b0f8723adf50aeb4051e1ca14(iPhone 12 Pro)

127 0 0 1_5173_neuron__u=s24we-diaaa-aaaaa-aaaka-cai neuron=9d49faba9fd2d31a5705db307bcef7ae0ef1ba2b0f8723adf50aeb4051e1ca14(iPhone 12 Pro) (2)

127 0 0 1_5173_neuron__u=s24we-diaaa-aaaaa-aaaka-cai neuron=9d49faba9fd2d31a5705db307bcef7ae0ef1ba2b0f8723adf50aeb4051e1ca14(iPhone 12 Pro) (3)

### CODE REVIEW

File: frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte [hunk 1]
```diff
@@ -22,6 +22,7 @@
   import NeuronStateRemainingTime from "$lib/components/neurons/NeuronStateRemainingTime.svelte";
   import { snsParametersStore } from "$lib/stores/sns-parameters.store";
   import type { Principal } from "@dfinity/principal";
+  import { secondsToDissolveDelayDuration } from "$lib/utils/date.utils.js";
```
Comment (peterpeterparker) [inline frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte:25]: `.js`?

File: frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte [hunk 2]
```diff
@@ -124,7 +125,15 @@
 
   <div>
     <p class="label">{$i18n.neurons.dissolve_delay_title}</p>
-    <p class="description">{$i18n.neurons.dissolve_delay_description}</p>
+    {#if minDissolveDelaySeconds !== undefined}
```
Comment (peterpeterparker) [inline frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte:128]: does the layout shift when `minDissolveDelaySeconds` is loaded?

if yes, would it be worth to always display the `<p>` and add the `#if` within?

Comment (peterpeterparker) [reply frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte:128]: also, should the test be `minDissolveDelaySeconds > 0`?

File: frontend/src/lib/utils/date.utils.ts [hunk 3]
```diff
@@ -48,6 +49,42 @@ export const secondsToDuration = (seconds: bigint): string => {
     .join(", ");
 };
 
+/**
+ * Displays years, months and days.
+ *
+ * Uses constants for `year` and `month`:
+ * - a year = 365.25 * 24 * 60 * 60
```
Comment (peterpeterparker) [inline frontend/src/lib/utils/date.utils.ts:56]: 365.25 to cover leap year, do I get it right?

File: frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte]: Didn't notice the shift locally but is updated.

File: frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte]: Thanks, fixed.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/sns-neurons/SetSnsDissolveDelay.svelte]: Thanks

File: frontend/src/lib/utils/date.utils.ts (hunk 3 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/date.utils.ts:56]: Coolio

**Pull request merged**